backend/src/main/java/com/bakdata/conquery/apiv1/FilterSearch.java

```diff
@@ -1,177 +1,278 @@
+	}
+
+	/**
+	 * To facilitate sharing/reuse among a dataset, we try to combine multiple columns into a single search:
+	 * If the column is part of a shared-Dictionary, we use that as reference.
+	 * Usually {@link Column}s of a {@link com.bakdata.conquery.models.datasets.SecondaryIdDescription} is also backed by a shared-Dictionary, but in the case it is not, we use the secondaryId as common reference name.
+	 * <p>
+	 * Lastly if the column is not shared or of a secondaryId we use the {@link com.bakdata.conquery.models.identifiable.ids.specific.ColumnId} itself.
+	 */
+	private static String decideColumnReference(Column column) {
+
+		if (column.getSharedDictionary() != null) {
+			return column.getSharedDictionary();
+		}
+
+		if (column.getSecondaryId() != null) {
+			return column.getSecondaryId().getId().toString();
+		}
+
+		return column.getId().toString();
 	}
 
-	private static Map<String, QuickSearch<FilterSearchItem>> search = new HashMap<>();
+	/**
+	 * For a {@link SelectFilter}, decide which references to use for searching.
+	 */
+	private static List<String> getSearchReferences(SelectFilter<?> filter) {
+		final List<String> references = new ArrayList<>(3);
+
+		if (filter.getTemplate() != null) {
+			references.add(filter.getTemplate().getFilePath());
+		}
+
+		references.add(filter.getId().toString());
+		references.add(decideColumnReference(filter.getColumn()));
+
+		return references;
+	}
 
 	/**
-	 * Scan all SelectFilters and submit {@link SimpleJob}s to create interactive searches for them.
+	 * For a {@link SelectFilter} collect all relevant {@link TrieSearch}.
 	 */
-	public static void updateSearch(DatasetRegistry datasets, Collection<Dataset> datasetsToUpdate, JobManager jobManager, CSVConfig parser) {
-		datasetsToUpdate.stream()
-				.flatMap(ds -> datasets.get(ds.getId()).getStorage().getAllConcepts().stream())
-				.flatMap(c -> c.getConnectors().stream())
-				.flatMap(co -> co.collectAllFilters().stream())
-				.filter(f -> f instanceof AbstractSelectFilter && ((AbstractSelectFilter<?>) f).getTemplate() != null)
-				.map(AbstractSelectFilter.class::cast)
-				.forEach(f -> jobManager.addSlowJob(new SimpleJob(String.format("SourceSearch[%s]", f.getId()), () -> createSourceSearch(f, parser))));
+	public List<TrieSearch<FEValue>> getSearchesFor(SelectFilter<?> filter) {
+		return getSearchReferences(filter).stream()
+										  .map(searchCache::get)
+										  .filter(Objects::nonNull)
+										  .collect(Collectors.toList());
 	}
 
-	/***
-	 * Create interactive Search for the selected filter based on its Template.
-	 * @param filter
+
+	/**
+	 * Scan all SelectFilters and submit {@link SimpleJob}s to create interactive searches for them.
 	 */
-	public static void createSourceSearch(AbstractSelectFilter<?> filter, CSVConfig parserConfig) {
-		FilterTemplate template = filter.getTemplate();
+	public void updateSearch(NamespaceStorage storage, JobManager jobManager, CSVConfig parser) {
```

Comment:
bisher sieht die die Klasse aus wie eine UtilityClass, nur dass `searchCache` einen state hat. Ich habs in einem anderen Kommentar noch geschrieben: Konstruktor über den die ganzen Namespace Dinge reingegeben werden.

Reply:
Gute Idee, das macht es ordentlicher.